src/storage/txn/actions/prewrite.rs

```diff
@@ -75,8 +75,15 @@ pub fn prewrite<S: Snapshot>(
         return Ok((min_commit_ts, OldValue::Unspecified));
     }
 
-    let old_value = if txn_props.need_old_value && mutation.mutation_type.may_have_old_value() {
-        if let Some(w) = prev_write {
+    let old_value = if txn_props.need_old_value {
```

Comment:
Seems you removed the `may_have_old_value` condition. I'm afraid it's not a good idea, since it makes `MutationType::Lock` also loads the old value.